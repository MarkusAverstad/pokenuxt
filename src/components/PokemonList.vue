<script setup lang="ts">
import type { FormattedPokemonSpecies } from '@types'

interface PokemonListProps {
  pokemonList?: FormattedPokemonSpecies[]
}

withDefaults(defineProps<PokemonListProps>(), {
  pokemonList: () => [] as FormattedPokemonSpecies[],
})
</script>

<template>
  <div>
    <div
      v-for="(pokemonSpecies, index) in (pokemonList)"
      :key="'pokemon_' + index"
      data-test="pokemon-card"
    >
      >
      <NuxtLink
        :to="`/pokemon/${pokemonSpecies.name}`"
      >
        {{ pokemonSpecies.formattedName }}
      </NuxtLink>
    </div>
  </div>
</template>

<style scoped>

</style>
