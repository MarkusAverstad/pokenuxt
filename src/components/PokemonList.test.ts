import { mount } from '@vue/test-utils'
import PokemonList from './PokemonList.vue'

describe('PokemonList', () => {
  it('should render', () => {
    const pokemonListWrapper = createWrapper()

    expect(pokemonListWrapper.exists()).toBe(true)
    expect(pokemonListWrapper.findAll('[data-test="pokemon-card"]')).toHaveLength(2)
    const bulbasaurLink = pokemonListWrapper.find('nuxt-link-stub[to="/pokemon/bulbasaur"]')
    const charmanderLink = pokemonListWrapper.find('nuxt-link-stub[to="/pokemon/charmander"]')
    expect(bulbasaurLink.exists()).toBe(true)
    expect(charmanderLink.exists()).toBe(true)
  })

  it('should match snapshot', () => {
    const pokemonListWrapper = createWrapper()

    expect(pokemonListWrapper.findAll('[data-test="pokemon-card"]')).toHaveLength(2)
    expect(pokemonListWrapper.html()).toMatchSnapshot()
  })

  const createWrapper = (props = {}) => {
    return mount(PokemonList, {
      props: {
        pokemonList: [
          {
            name: 'bulbasaur',
            formattedName: 'Bulbasaur',
            url: 'https://pokeapi.co/api/v2/pokemon/1/',
          },
          {
            name: 'charmander',
            formattedName: 'Charmander',
            url: 'https://pokeapi.co/api/v2/pokemon/4/',
          },
        ],
        ...props,
      },
      global: {
        stubs: {
          NuxtLink: true,
        },
      },
    })
  }
})
