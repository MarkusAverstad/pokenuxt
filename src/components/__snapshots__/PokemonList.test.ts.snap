// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonList > should match snapshot 1`] = `
"<div>
  <div data-test="pokemon-card"> &gt; <nuxt-link-stub to="/pokemon/bulbasaur"></nuxt-link-stub>
  </div>
  <div data-test="pokemon-card"> &gt; <nuxt-link-stub to="/pokemon/charmander"></nuxt-link-stub>
  </div>
</div>"
`;
